internal/db/schema/migrations/postgres/11/01_server_type_enum_test.go

```diff
@@ -0,0 +1,191 @@
+	ctx := context.Background()
+
+	c, u, _, err := docker.StartDbInDocker(dialect)
+	require.NoError(err)
+	t.Cleanup(func() {
+		require.NoError(c())
+	})
+	d, err := sql.Open(dialect, u)
+	require.NoError(err)
+
+	// migration to the prior migration (before the one we want to test)
+	oState := schema.TestCloneMigrationStates(t)
+	nState := schema.TestCreatePartialMigrationState(oState["postgres"], priorMigration)
+	oState["postgres"] = nState
+
+	m, err := schema.NewManager(ctx, dialect, d, schema.WithMigrationStates(oState))
+	require.NoError(err)
+
+	require.NoError(m.RollForward(ctx))
+	state, err := m.CurrentState(ctx)
+	require.NoError(err)
+	require.Equal(priorMigration, state.DatabaseSchemaVersion)
+	require.False(state.Dirty)
+
+	// okay, now we can seed the database with test data
+	conn, err := gorm.Open(dialect, u)
+	require.NoError(err)
+	rootWrapper := db.TestWrapper(t)
+	repo := servers.TestRepo(t, conn, rootWrapper)
+
+	// Seed the data
+	origController := &servers.Server{
+		PrivateId: "test-controller",
+		Type:      "controller",
+		Address:   "127.0.0.1",
+	}
+	_, _, err = repo.UpsertServer(ctx, origController)
+	require.NoError(err)
+
+	origWorker := &servers.Server{
+		PrivateId: "test-worker",
+		Type:      "worker",
+		Address:   "127.0.0.1",
+		Tags: map[string]*servers.TagValues{
+			"tag": {
+				Values: []string{"value1", "value2"},
+			},
+		},
+	}
+	_, _, err = repo.UpsertServer(ctx, origWorker, servers.WithUpdateTags(true))
+	require.NoError(err)
+
+	// Read the servers back
+	expectedControllers, err := repo.ListServers(ctx, servers.ServerTypeController)
+	require.NoError(err)
+	require.Len(expectedControllers, 1)
+
+	expectedWorkers, err := repo.ListServers(ctx, servers.ServerTypeWorker)
+	require.NoError(err)
+	require.Len(expectedWorkers, 1)
+
+	// now we're ready for the migration we want to test.
+	oState = schema.TestCloneMigrationStates(t)
+	nState = schema.TestCreatePartialMigrationState(oState["postgres"], serverEnumMigration)
+	oState["postgres"] = nState
+
+	m, err = schema.NewManager(ctx, dialect, d, schema.WithMigrationStates(oState))
+	require.NoError(err)
+
+	require.NoError(m.RollForward(ctx))
+	state, err = m.CurrentState(ctx)
+	require.NoError(err)
+	require.Equal(serverEnumMigration, state.DatabaseSchemaVersion)
+	require.False(state.Dirty)
+
+	t.Log("current migration version: ", state.DatabaseSchemaVersion)
+
+	// Assert servers
+	actualControllers, err := repo.ListServers(ctx, servers.ServerTypeController)
+	require.NoError(err)
+	require.Len(actualControllers, 1)
+	require.Equal(expectedControllers, actualControllers)
+
+	actualWorkers, err := repo.ListServers(ctx, servers.ServerTypeWorker)
+	require.NoError(err)
+	require.Len(actualWorkers, 1)
+	require.Equal(expectedWorkers, actualWorkers)
+
+	// Assert the state of the newly created enum table
+	rows, err := d.QueryContext(ctx, "select * from server_type_enm")
+	require.NoError(err)
+
+	// Column types
+	colTypes, err := rows.ColumnTypes()
```

Comment:
I'm not sure we need to test that the DDL is correct, given that you have a FK that ensures it's type and name as well.  The same is likely true of the inserted data.

Reply:
I agree. It feels like an anti-pattern to start testing the DDL.